dev/AppLifecycle/ActivationRegistrationManager.cpp

```diff
@@ -58,39 +100,75 @@ namespace winrt::Microsoft::ProjectReunion::implementation
             throw hresult_illegal_method_call();
         }
 
-        RegisterProtocol(scheme.c_str());
+        RegisterForProtocolActivationInternal(schemeName.c_str(), L"", logo.c_str(),
+            displayName.c_str(), exePath.c_str());
+    }
+
+    void ActivationRegistrationManager::UnregisterForProtocolActivation(hstring const& scheme,
+        hstring const& exePath)
+    {
+        if (scheme.empty())
+        {
+            throw hresult_invalid_argument();
+        }
+
+        if (HasIdentity())
+        {
+            throw hresult_illegal_method_call();
+        }
 
-        auto appId = ComputeAppId();
+        auto appId = ComputeAppId(exePath.c_str());
         auto type = AssociationType::Protocol;
         auto progId = ComputeProgId(appId, type);
+        UnregisterAssociationHandler(appId, scheme.c_str(), type);
+        UnregisterProgId(progId);
+    }
+
+    void ActivationRegistrationManager::RegisterForStartupActivation(hstring const& taskId,
+        hstring const& exePath)
+    {
+        if (HasIdentity())
+        {
+            throw hresult_illegal_method_call();
+        }
 
-        RegisterProgId(progId.c_str(), L"", applicationDisplayName.c_str(), logo.c_str());
+        // Example: HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Run
+        wil::unique_hkey key;
+        THROW_IF_WIN32_ERROR(::RegOpenKeyEx(HKEY_CURRENT_USER, c_runKeyPath.c_str(), 0, KEY_WRITE,
```

Comment:
>KEY_WRITE [](start = 88, length = 9)

Any concerns about WOW? e.g. x86 process calls this on an x64 machine.

Reply:
No, this key is not virtualized by WOW.